### ISSUE TITLE
Hat-Guessing for Throw It in a Hole

### ISSUE THREAD

**Issue by iamwhoiamhahaha** [Zamiell/hanabi-conventions#180]
JustWhy?Today at 1:18 AM
> Hat-guessing has high efficiency, high tempo and high safety. But it has low information factor.

JustWhy?Today at 1:27 AM
> Hat guessing's weak points are variants where you either have to know more about your hand (such as throw it in a hole variant) or where you can't send many 'commands' (e.g. duck).

IAMJEFFToday at 1:31 AM
> throw it in a hole is definitely not hat-guessingable but i believe we can find a way to hat-guessing duck

JustWhy?Today at 1:37 AM
> Probably, if you clue at least one-away player and the next player sees how many options was there for previous player to clue.
And then decides what the 'command' is.

IAMJEFFToday at 1:42 AM
> i'll make a table for duck after my advanced hat-guessing conventions got approved
need to find a way to beat duck 3 suits 5-player

JustWhy?Today at 1:59 AM
> Plus, there could still be at least 2 types of commands the next player can be clued. A clue that touches next player's chop and a clue that doesn't.

ZamielToday at 2:54 AM
> @Florrat justwhy has a really good point, hat guessing sucks with throw it in a hole. what would be the best possible algorithm to solve that? some new system entirely?
you'd need to encode card identities

JustWhy?Today at 3:04 AM
> Even more information-dependent variant would be the one where you have to drag a card onto a correct stack.

FlorratToday at 9:55 AM
> Yes, hat guessing has to be adapted to use it for throw it in a hole.
If you allow empty clues, it's still pretty easy with 4/5 players: in that case there are more clues you can give than there are cards. For example with 4 player Throw it in a hole (6 Suit) you can give 33 clues, and there are 30 cards. Then you can make a table modulo 30 for all cards, and you hint to each player their oldest unknown card. This idea was described to me here: https://github.com/SashaIr/hanabi-conventions/pull/1#issuecomment-603184057

> Without empty clues you can do something similar, but it's weaker and trickier. For example with 5 player Throw it in a hole (6 Suit) you can easily distinguish 16 clues (number/color that does/doesn't touch newest on player X). You can still encode every player's oldest unkown card by telling them
playable card of color X (6 possibilities), 1-away card of color X (6 possibilities), any 2-away card, any 3-away card, any 4-away card, trash. If someone has some information about their whole hand, you can exactly identify one of the cards that was only partially identified before. This should work reasonably well.

ZamielToday at 10:20 AM
> actually you only need to tell them the suit right?
because if they are playing a suit then they can derive the rank

FlorratToday at 10:28 AM
> no, you're telling everyone their oldest unknown card, independent of whether that card is playable or not.

ZamielToday at 10:30 AM
> i dont see why that is good

MasNToday at 10:30 AM
> is that the same idea as this one? https://boardgamearena.com/forum/viewtopic.php?f=102&t=11865

ZamielToday at 10:41 AM
> 
> ```
> | # mod 25 | action        | person clued   | type of clue
> | -------- | ------------- | -------------- | -------------
> | 0        | give clue     | 1 player away  | number on newest card
> | 1        | play 1 red    | 1 player away  | color on newest card
> | 2        | play 2 red    | 1 player away  | any clue not touching the newest card
> | 3        | play 3 red    | 2 player away  | number on newest card
> | 4        | play 4 red    | 2 players away | color on newest card
> | 5        | play 1 yellow | 1 player away  | color on newest card
> | 6        | play 2 yellow | 1 player away  | any clue not touching the newest card
> | 7        | play 3 yellow | 2 player away  | number on newest card
> | 8        | play 4 yellow | 2 players away | color on newest card
> | 9        | play 1 green  | 1 player away  | color on newest card
> | 10       | play 2 green  | 1 player away  | any clue not touching the newest card
> | 11       | play 3 green  | 2 player away  | number on newest card
> | 12       | play 4 green  | 2 players away | color on newest card
> | 13       | play 1 blue   | 1 player away  | color on newest card
> | 14       | play 2 blue   | 1 player away  | any clue not touching the newest card
> | 15       | play 3 blue   | 2 player away  | number on newest card
> | 16       | play 4 blue   | 2 players away | color on newest card
> | 17       | play 1 purple | 1 player away  | color on newest card
> | 18       | play 2 purple | 1 player away  | any clue not touching the newest card
> | 19       | play 3 purple | 2 player away  | number on newest card
> | 20       | play 4 purple | 2 players away | color on newest card
> | 21       | discard 1     | 2 players away | any clue not touching the newest card
> | 22       | discard 2     | 3 players away | number on newest card
> | 23       | discard 3     | 3 players away | color on newest card
> | 24       | discard 4     | 3 players away | any clue not touching the newest card
> ```
> 

> ignore "type of clue" column for now
is that good?
basically mod 30 is nice but only mod 25 should be "necessary"
maybe we can slim it down by arbitrarily removing slot 2 and/or slot 3?

IAMJEFFToday at 10:58 AM
> ohh that sounds great

**Comment by iamwhoiamhahaha** [comment]
> 
> ```
> | # mod 16 | action        | person clued   | type of clue
> | -------- | ------------- | -------------- | -------------
> | 0        | give clue     | 1 player away  | number on newest card
> | 1        | play 1 red    | 1 player away  | color on newest card
> | 2        | play 3 red    | 1 player away  | number not touching the newest card
> | 3        | play 4 red    | 1 player away  | color not touching the newest card
> | 4        | play 1 yellow | 2 player away  | number on newest card
> | 5        | play 3 yellow | 2 player away  | color on newest card
> | 6        | play 4 yellow | 2 players away | number not touching the newest card
> | 7        | play 1 green  | 2 player away  | color not touching the newest card
> | 8        | play 3 green  | 3 player away  | number on newest card
> | 9        | play 4 green  | 3 players away | color on newest card
> | 10       | play 1 blue   | 3 player away  | number not touching the newest card
> | 11       | play 3 blue   | 3 player away  | color not touching the newest card
> | 12       | play 4 blue   | 4 players away | number on newest card
> | 13       | play 1 purple | 4 player away  | color on newest card
> | 14       | play 3 purple | 4 player away  | number not touching the newest card
> | 15       | play 4 purple | 4 players away | color not touching the newest card
> | 16       | chop move     | N/A            | discard your chop
> ```

**Comment by Zamiell** [comment]
seems fine

**Comment by iamwhoiamhahaha** [comment]
### Hat-Guessing for Throw It in a Hole

### Clue Interpretation Tables (5 Suits, 5-player)
 
 ```
 | # mod 16 | action        | person clued   | type of clue
 | -------- | ------------- | -------------- | -------------
 | 0        | give clue     | 1 player away  | number on newest card
 | 1        | play 1 red    | 1 player away  | color on newest card
 | 2        | play 3 red    | 1 player away  | number not touching the newest card
 | 3        | play 4 red    | 1 player away  | color not touching the newest card
 | 4        | play 1 yellow | 2 player away  | number on newest card
 | 5        | play 3 yellow | 2 player away  | color on newest card
 | 6        | play 4 yellow | 2 players away | number not touching the newest card
 | 7        | play 1 green  | 2 player away  | color not touching the newest card
 | 8        | play 3 green  | 3 player away  | number on newest card
 | 9        | play 4 green  | 3 players away | color on newest card
 | 10       | play 1 blue   | 3 player away  | number not touching the newest card
 | 11       | play 3 blue   | 3 player away  | color not touching the newest card
 | 12       | play 4 blue   | 4 players away | number on newest card
 | 13       | play 1 purple | 4 player away  | color on newest card
 | 14       | play 3 purple | 4 player away  | number not touching the newest card
 | 15       | play 4 purple | 4 players away | color not touching the newest card
 | 16       | chop move     | N/A            | discard your chop
 ```

### Clue Interpretation Tables (4 Suits, 5-player)
 
 ```
 | # mod 15 | action        | person clued   | type of clue
 | -------- | ------------- | -------------- | -------------
 | 0        | give clue     | 1 player away  | number on newest card
 | 1        | play 1 red    | 1 player away  | color on newest card
 | 2        | play 3 red    | 1 player away  | number not touching the newest card
 | 3        | play 4 red    | 1 player away  | color not touching the newest card
 | 4        | play 1 yellow | 2 player away  | number on newest card
 | 5        | play 3 yellow | 2 player away  | color on newest card
 | 6        | play 4 yellow | 2 players away | number not touching the newest card
 | 7        | play 1 green  | 2 player away  | color not touching the newest card
 | 8        | play 3 green  | 3 player away  | number on newest card
 | 9        | play 4 green  | 3 players away | color on newest card
 | 10       | play 1 blue   | 3 player away  | number not touching the newest card
 | 11       | play 3 blue   | 3 player away  | color not touching the newest card
 | 12       | play 4 blue   | 4 players away | number on newest card
 | 13       | discard 2     | 4 player away  | color on newest card
 | 14       | discard 3     | 4 player away  | number not touching the newest card
 | 15       | discard 4     | 4 players away | color not touching the newest card
 ```

### Clue Interpretation Tables (3 Suits, 5-player)
 
 ```
 | # mod 15 | action        | person clued   | type of clue
 | -------- | ------------- | -------------- | -------------
 | 0        | give clue     | 1 player away  | number on newest card
 | 1        | play 1 red    | 1 player away  | color on newest card
 | 2        | play 2 red    | 1 player away  | number not touching the newest card
 | 3        | play 3 red    | 1 player away  | color not touching the newest card
 | 4        | play 4 red    | 2 player away  | number on newest card
 | 5        | play 1 yellow | 2 player away  | color on newest card
 | 6        | play 2 yellow | 2 players away | number not touching the newest card
 | 7        | play 3 yellow | 2 player away  | color not touching the newest card
 | 8        | play 4 yellow | 3 player away  | number on newest card
 | 9        | play 1 green  | 3 players away | color on newest card
 | 10       | play 2 green  | 3 player away  | number not touching the newest card
 | 11       | play 3 green  | 3 player away  | color not touching the newest card
 | 12       | play 4 green  | 4 players away | number on newest card
 | 13       | discard 2     | 4 player away  | color on newest card
 | 14       | discard 3     | 4 player away  | number not touching the newest card
 | 15       | discard 4     | 4 players away | color not touching the newest card
 ```

**Comment by iamwhoiamhahaha** [comment]
### Clue Interpretation Tables (6 Suits, 5-player)
 
 ```
 | # mod 16 | action        | person clued   | type of clue
 | -------- | ------------- | -------------- | -------------
 | 0        | give clue     | 1 player away  | number on newest card
 | 1        | play 2 red    | 1 player away  | color on newest card
 | 2        | play 4 red    | 1 player away  | number not touching the newest card
 | 3        | play 2 yellow | 1 player away  | color not touching the newest card
 | 4        | play 4 yellow | 2 player away  | number on newest card
 | 5        | play 2 green  | 2 player away  | color on newest card
 | 6        | play 4 green  | 2 players away | number not touching the newest card
 | 7        | play 2 blue   | 2 player away  | color not touching the newest card
 | 8        | play 4 blue   | 3 player away  | number on newest card
 | 9        | play 2 purple | 3 players away | color on newest card
 | 10       | play 4 purple | 3 player away  | number not touching the newest card
 | 11       | play 2 teal   | 3 player away  | color not touching the newest card
 | 12       | play 4 teal   | 4 players away | number on newest card
 | 13       | discard 2     | 4 player away  | color on newest card
 | 14       | discard 3     | 4 player away  | number not touching the newest card
 | 15       | discard 4     | 4 players away | color not touching the newest card
 ```

**Comment by Zamiell** [comment]
isn't slot 1 most likely to be playable?

**Comment by iamwhoiamhahaha** [comment]
if someone has playable card on slot 2, and we need to wait until that card is on slot 4

**Comment by iamwhoiamhahaha** [comment]
### Clue Interpretation Tables (3 Suits, 5-player)
 
 ```
 | # mod 16 | action          | person clued   | type of clue
 | -------- | --------------- | -------------- | -------------
 | 0        | give clue       | 1 player away  | number on newest card
 | 1        | play 1 red      | 1 player away  | color on newest card
 | 2        | play 2 red      | 1 player away  | number not touching the newest card
 | 3        | play 3 red      | 1 player away  | color not touching the newest card
 | 4        | play 4 red      | 2 player away  | number on newest card
 | 5        | play 1 yellow   | 2 player away  | color on newest card
 | 6        | play 2 yellow   | 2 players away | number not touching the newest card
 | 7        | play 3 yellow   | 2 player away  | color not touching the newest card
 | 8        | play 4 yellow   | 3 player away  | number on newest card
 | 9        | play 1 green    | 3 players away | color on newest card
 | 10       | play 2 green    | 3 player away  | number not touching the newest card
 | 11       | play 3 green    | 3 player away  | color not touching the newest card
 | 12       | play 4 green    | 4 players away | number on newest card
 | 13       | play 1 one-away | 4 player away  | color on newest card
 | 14       | play 3 one-away | 4 player away  | number not touching the newest card
 | 15       | play 4 one-away | 4 players away | color not touching the newest card
 ```
Also apply [Three Suits 5-player Conventions](https://github.com/Zamiell/hanabi-conventions/issues/168)

> Notes: Playing one-away cards might be a problem but we need efficiency somehow.

**Comment by iamwhoiamhahaha** [comment]
ohh we dont have throw it in a hole 3 suits

**Comment by Zamiell** [comment]
did you update your guide with this information? can I close this issue?

**Comment by iamwhoiamhahaha** [comment]
yes
